src/OnChain.ts

```diff
@@ -86,12 +87,22 @@ export const OnChainMixin = (superclass) =>
       if (payeeUser) {
         fee = 0
       } else {
+        if (amount && amount < yamlConfig.onchainDustAmount) {
+          throw new DustAmountError(undefined, { logger: this.logger })
```

Comment:
This might be a good place to start introducing a new way to return errors via a union type like discussed in code review.